core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -642,9 +672,95 @@ public CompletableFuture<Optional<Throwable>> acknowledge(String memberId, List<
             lock.writeLock().unlock();
         }
 
+        if (canMoveStartOffset) {
+            maybeUpdateCachedStateAndOffsets();
+        }
+
         return CompletableFuture.completedFuture(Optional.ofNullable(throwable));
     }
 
+    // Visible for testing
+    public long startOffset() {
+        return this.startOffset;
+    }
+
+    // Visible for testing
+    public long endOffset() {
+        return this.endOffset;
+    }
+
+    public boolean canAcquireMore() {
```

Comment:
Would be good to structure the code with Public -> Protected -> Private -> Visible for Testing methods for readability. Also please add javadoc for the public method.